x-pack/platform/plugins/shared/ml/server/routes/data_frame_analytics.ts

```diff
@@ -470,9 +471,25 @@ export function dataFrameAnalyticsRoutes(
                 // If user does have privilege to delete the index, then delete the index
                 if (userCanDeleteDestIndex) {
                   try {
-                    await client.asCurrentUser.indices.delete({
+                    // It's possible that the destination index has been reindexed with a new name
+                    // so if it's an alias first, then delete the real index
+                    let destinationIndexToDelete = destinationIndex;
+                    const alias = await client.asCurrentUser.indices.getAlias({
                       index: destinationIndex,
                     });
+                    if (alias) {
+                      const reindexedDestName = Object.keys(alias)[0];
+                      if (reindexedDestName && isPopulatedObject(alias, [reindexedDestName])) {
+                        const keys = Object.keys(alias[reindexedDestName]?.aliases);
+                        if (keys[0] === destinationIndex) {
+                          destinationIndexToDelete = reindexedDestName;
+                        }
+                      }
+                    }
+
+                    await client.asCurrentUser.indices.delete({
```

Comment:
The alias will need to be deleted too I believe.

Reply:
When I was testing this, deleting the index makes the alias not found since it's a one-to-one correlation. But I'll double check again and add in if needed.